File: client/src/lib/server/artwork.ts
import type { Artwork } from '../../types/artwork';

export async function fetchArtworks(page: string): Promise<Artwork[]> {
	const res = await fetch('http://localhost:5173/api/artworks?page=' + page);
	if (!res.ok) {
		throw new Error('Failed to fetch artworks from the server (HTTP ' + res.status + ')');
	}
	return res.json();
}

File: api/internal/handlers/artwork.go
package handlers

import (
	"api/internal/services"
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
)

func GetArtworks(c *gin.Context) {
	pageStr := c.DefaultQuery("page", "1")
	limitStr := c.DefaultQuery("limit", "25")

	page, err := strconv.Atoi(pageStr)
	if err != nil || page < 1 {
		page = 1
	}

	limit, err := strconv.Atoi(limitStr)
	if err != nil || limit < 1 {
		limit = 10
	}

	artworks, err := services.FetchArtworks(page, limit)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, artworks)
}
